verilog/tools/ls/verilog-language-server_test.cc

```diff
@@ -1442,6 +1442,138 @@ TEST_F(VerilogLanguageServerSymbolTableTest, CheckReferenceUnknownSymbol) {
   ASSERT_EQ(response_b["result"].size(), 0);
 }
 
+struct RenameRequestParams {
+  int id;
+  std::string file;
+  std::string newName;
+  int line;
+  int character;
+};
+
+std::string RenameRequest(RenameRequestParams params) {
+  json renamerq = {
+
+      {"jsonrpc", "2.0"},
+      {"id", params.id},
+      {"method", "textDocument/rename"},
+      {"params",
+       {
+           {"textDocument",
+            {
+                {"uri", params.file},
+            }},
+           {
+               "position",
+               {{"line", params.line}, {"character", params.character}},
+           },
+           {"newName", params.newName},
+       }}};
+  return renamerq.dump();
+}
+std::string PrepareRenameRequest(RenameRequestParams params) {
+  json renamerq = {
+
+      {"jsonrpc", "2.0"},
+      {"id", params.id},
+      {"method", "textDocument/prepareRename"},
+      {"params",
+       {
+           {"textDocument",
+            {
+                {"uri", params.file},
+            }},
+           {
+               "position",
+               {{"line", params.line}, {"character", params.character}},
+           },
+           {"newName", params.newName},
+       }}};
+  return renamerq.dump();
+}
+
+// Runs tests for textDocument/rangeFormatting requests
+TEST_F(VerilogLanguageServerSymbolTableTest, PrepareRenameTest) {
+  // Create sample file and make sure diagnostics do not have errors
+  RenameRequestParams params;
```

Comment:
Given that we're here in a test that figures out prepare rename, having the parameter called `RenameRequestParams` is of course confusing; also a good reason to switch to the auto-generated protocol structs (`verible::lsp::PrepareRenameParams`).